packages/tracing-internal/src/browser/browsertracing.ts

```diff
@@ -329,7 +329,7 @@ export class BrowserTracing implements Integration {
       const { idleTimeout, finalTimeout, heartbeatInterval } = this.options;
 
       const op = 'ui.action.click';
-      if (inflightInteractionTransaction) {
+      if (inflightInteractionTransaction && !inflightInteractionTransaction.endTimestamp) {
```

Comment:
calling finish twice is fine though because we have guards to make sure we don't flush twice

 https://github.com/getsentry/sentry-javascript/blob/120ae6de8b918fd617691d43bec48083f8066878/packages/core/src/tracing/transaction.ts#L132